zio-interop-cats/shared/src/main/scala/zio/interop/cats.scala

```diff
@@ -237,8 +242,8 @@ private class CatsConcurrent[R] extends CatsMonadError[R, Throwable] with Concur
       fc.interruptible
 
     (run(fa) raceWith run(fb))(
-      { case (l, f) => l.fold(f.interrupt *> ZIO.failCause(_), ZIO.succeedNow).map(lv => Left((lv, toFiber(f)))) },
-      { case (r, f) => r.fold(f.interrupt *> ZIO.failCause(_), ZIO.succeedNow).map(rv => Right((toFiber(f), rv))) }
+      { case (l, f) => l.foldExit(f.interrupt *> ZIO.failCause(_), ZIO.succeedNow).map(lv => Left((lv, toFiber(f)))) },
+      { case (r, f) => r.foldExit(f.interrupt *> ZIO.failCause(_), ZIO.succeedNow).map(rv => Right((toFiber(f), rv))) }
```

Comment:
curious here as to what the difference is between `foldExit` and `foldCauseZIO` since they both compile and run with the same test results (looks like foldExit might save an instantiation?)

Reply:
`foldExit` preserves the type information that the value is an `Exit` and not some other subtype of `ZIO`.

Reply:
ah okay, i missed that Exit is a ZIO and that foldExit can only be called on an Exit...but since it's a fold with inputs being a Cause or an A I don't see how it preserves type information. but probably something i'm missing, not asking for an explanation at midnight.

Reply:
`foldCauseZIO` returns a `ZIO` versus `foldExit` returns an `Exit`. So if use `foldCauseZIO` we "forget" that this is a workflow that is "done" and we can just get the value from.

Reply:
Also notice how `foldExit` takes a function that returns a `Z` and returns a `Z` and `foldCauseZIO` takes a function that returns a `ZIO` and returns a `ZIO`. So they are going to be the same when you are returning a `ZIO` but `foldExit` allows you to actually get a value out of a `ZIO` in the case where it is an `Exit`.